java/README.md

```diff
@@ -1,14 +1,42 @@
-Prerequisites
--------------
+# CuVS Java API
 
-* JDK 22
-* Maven 3.9.6 or later
+CuVS Java API attempts to provide a Java based simple, efficient, and a robust vector search API.
 
-To build this API, please do `./build.sh java` in the top level directory. Since this API is dependent on `libcuvs` it must be noted that `libcuvs` gets built automatically before building this API.
+> [!CAUTION]
+> This is an experimental release and updates to this API should be expected in the coming days.
 
-Alternatively, please build libcuvs (`./build.sh libcuvs` from top level directory) before building the Java API with `./build.sh` from this directory.
 
-Building
---------
+## Prerequisites
 
-`./build.sh` will generate the `libcuvs_java.so` file in the `internal/` directory, and then build the final jar file for the cuVS Java API in the `cuvs-java/` directory.
+- gcc 11.4
+- [nvcc 12.4 or above](https://developer.nvidia.com/cuda-downloads)
+- [cmake 3.28 or above](https://cmake.org/download/)
+- [maven 3.9.6 or above](https://maven.apache.org/download.cgi)
+- [JDK 22](https://openjdk.org/projects/jdk/22/)
+- [Ubuntu 22.04](https://releases.ubuntu.com/jammy/)
+
+
+## Building
+
+libcuvs libraries are needed for this API. If libcuvs libraries are already not built please do `./build.sh libcuvs java` in the top level directory to build this API.
```

Comment:
I'm really not a fan of having this embedded in the java directory. I'd much prefer that this information is provided directly in the `build cuVS from source` docs and this README just links to that. It just simply makes things hard to find, maintain (and update) when this info is scattered around the codebase.

Let's keep it here for the moment while this API is in experimental state, but plan to move into the docs in 25.04 release cycle. 

```suggestion
The libcuvs C and C++ libraries are needed for this API. If libcuvs libraries have not been built and installed, use `./build.sh libcuvs java` in the top level directory to build this API.
```